### ISSUE TITLE
Add CMake option to disable installing

### ISSUE THREAD

**Issue by nickanthony-dgl** [troldal/OpenXLSX#239]
It would be nice to have an option like "OPENXLSX_ENABLE_INSTALL" that allows projects using this as  subroject to avoid having the headers and static libs installed.

**Comment by nickanthony-dgl** [comment]
I am doing the same as what you descirbe above with the addition of `install(TARGETS your_exe)`. I want to be able to configure cmake so that when I run `cmake --install` it will install my targets (i.e. `your_exe`) but it won't also install extra things like OpenXLSX's targets.

So I am proposing that OpenXLSX adds something like `option(OPENXLSX_ENABLE_INSTALL "Should headers and static lib be installed" ON)`. Then you use `if(OPENXLSX_ENABLE_INSTALL)` to guard any part of OpenXLSX CMake files that use the `install()` command. Then people using the library with `add_subdirectory` just need to set this option to OFF if they don't want to install OpenXLSXs targets.

Many libraries do the same thing. For example take a look at GTest's `INSTALL_GTEST` option https://github.com/google/googletest/blob/main/CMakeLists.txt

**Comment by jurassicLizard** [comment]
Yeah, i can see many cases where that is needed . If you are interested ,  i currently tested with a rather small, private project and it appears using the EXCLUDE_FROM_ALL might achieve exactly what you want by excluding OpenXLSXs own install targets and only linking against it (tested with CMAKE 3.16.3)

 ```cmake
add_subdirectory(relative/path/to/OpenXLSX EXCLUDE_FROM_ALL)
```  



Source : https://cmake.cmake.narkive.com/WKIKLMMB/disabling-install-target-for-subdirectory

**Comment by nickanthony-dgl** [comment]
I'll give that a try, thanks.

**Comment by aral-matrix** [comment]
> I'll give that a try, thanks.

Since I am finally understanding a little more about how cmake works, I have to concur with the suggestion of @jurassicLizard - the ```EXCLUDE_FROM_ALL``` flag should fix your issue, @nickanthony-dgl  - did you have success with your try? If you could report back that would be great.
If I don't hear anything, I will close this issue as resolved within the next two weeks.

**Comment by nickanthony-dgl** [comment]
Yes, sorry I should've gotten back to you on this. EXLCUDE_FROM_ALL seems to do exactly what I needed.

**Comment by aral-matrix** [comment]
Okay, that's great to hear. Thank you for the feedback! :)
